### DISCUSSION TITLE
Project Layout [Best Practices]

### DISCUSSION THREAD

**Discussion by dantium** [litestar-org/litestar#821] [Ideas]
Like other opinionated frameworks such as Django it maybe a good idea to have some strong recommendations for project layout, directory structure, naming conventions etc. 

Best practices would give consistency and be helpful for beginners. Conventions should be used in tutorials, sample applications and project/app scaffolding via the CLI #603

**Comment by rhacer63** [comment]
My layout is inferred from the sample application in the docs, with a few additions for tests and VSCode.

![image](https://user-images.githubusercontent.com/27306556/202719769-c0e39929-7859-4338-9346-5a27d6769c6a.png)

**Reply by kasir-barati** [reply]
I think we really need a better project structure: https://github.com/orgs/litestar-org/discussions/1013

**Comment by peterschutt** [comment]
I've been working on an application configuration library which is getting close to stable: https://topsport-com-au.github.io/starlite-saqlalchemy/v0.12/

And I've just pushed up an example application using the library: https://github.com/topsport-com-au/example-starlite-saqlalchemy

Still some work to do on both, especially in terms of building in authn and authz to the pattern, and documentation, but I'll be chipping away at that (with one eye on the upcoming `starlite-security` library!). 

I also hope to put together a cookie cutter template based on the pattern soon.

**Reply by kasir-barati** [reply]
Why NOT a feature-modular project structure?

Discussed also here: https://github.com/orgs/litestar-org/discussions/1013
